File: 6kyu/Consecutivestrings_v1.ts

              // Consecutivestrings
              // https://www.codewars.com//kata/56a5d994ac971f1ac500003e
        
              export class G964 {
    public static longestConsec(strarr:string[], k:number):string {
        if(strarr.length === 0 || k > strarr.length || k <= 0) {
          return "";
        }
  
        let nums:number[] = strarr.map(x => x.length);
        let results:number[] = [];
        let num:number = 0;
  
        for(let i = 0; i < (nums.length - k + 1); i++){
          num = 0;
          for(let j = 0; j < k; j++){
            num += nums[i+j];
          }
          results.push(num);
        }
  
        let index:number = 0;
  
        for(let i = 0; i < results.length; i++){
          if(results[i] > results[index]){
            index = i;
          }
        }
  
        let str:string = "";
  
        for(let i = index; i < (index + k); i++){
          str += strarr[i];
        }
  
        return str;
    }
}

            

File: 6kyu/Consecutivestrings_v1.go

              // Consecutivestrings
              // https://www.codewars.com//kata/56a5d994ac971f1ac500003e
        
              package kata

func LongestConsec(strarr []string, k int) string {
    if(len(strarr) == 0 || k > len(strarr) || k <= 0) {
    return ""
  }
  
  nums := make([]int, 0)
  for _, st := range strarr {
    nums = append(nums, len(st))
  }
  
  results := make([]int, 0)
  num := 0;
  
  for i := 0; i < (len(nums) - k + 1); i++ {
    num = 0
    for j := 0; j < k; j++ {
      num += nums[i+j];
    }
    results = append(results, num);
  }
  
  index := 0;
  
  for i := 0; i < len(results); i++ {
    if results[i] > results[index] {
      index = i;
    }
  }
  
  str := "";
  
  for i := index; i < (index + k); i++ {
    str += strarr[i];
  }
  
  return str;
}
            